src/inputs/TextFieldBase.tsx

```diff
@@ -170,114 +170,129 @@ export function TextFieldBase<X extends Only<TextFieldXss, X>>(props: TextFieldB
-                  // Reset focus to input element
-                  fieldRef.current?.focus();
+              {...hoverProps}
+              ref={inputWrapRef as any}
+            >
+              {!multiline && labelStyle === "inline" && label && (
+                <InlineLabel labelProps={labelProps} label={label} {...tid.label} />
+              )}
+              {!multiline && startAdornment && <span css={Css.df.aic.fs0.br4.pr1.$}>{startAdornment}</span>}
+              <ElementType
+                {...mergeProps(
+                  inputProps,
+                  { onBlur, onFocus: onFocusChained, onChange: onDomChange },
+                  { "aria-invalid": Boolean(errorMsg), ...(labelStyle === "hidden" ? { "aria-label": label } : {}) },
+                )}
+                {...(errorMsg ? { "aria-errormessage": errorMessageId } : {})}
+                ref={fieldRef as any}
+                rows={multiline ? 1 : undefined}
+                css={{
+                  ...fieldStyles.input,
+                  ...(inputProps.disabled ? fieldStyles.disabled : {}),
+                  ...(showHover ? fieldStyles.hover : {}),
+                  ...(multiline ? Css.h100.p1.add("resize", "none").$ : Css.truncate.$),
+                  ...xss,
                 }}
+                {...tid}
               />
+              {isFocused && clearable && onChange && inputProps.value && (
+                <IconButton
+                  icon="xCircle"
+                  color={Palette.Gray700}
+                  onClick={() => {
+                    onChange(undefined);
+                    // Reset focus to input element
+                    fieldRef.current?.focus();
+                  }}
+                />
+              )}
+              {errorInTooltip && errorMsg && !hideErrorMessage && (
+                <span css={Css.df.aic.pl1.fs0.$}>
+                  <Icon icon="error" color={Palette.Red600} tooltip={errorMsg} />
+                </span>
+              )}
+              {!multiline && endAdornment && <span css={Css.df.aic.pl1.fs0.$}>{endAdornment}</span>}
+            </div>
+          ),
+        })}
+        {/* Compound fields will handle their own error and helper text. Do not show error or helper text when 'readOnly' or disabled */}
+        {labelStyle !== "left" && !compound && !inputProps.disabled && !inputProps.readOnly && (
+          <>
+            {errorMsg && !errorInTooltip && (
+              <ErrorMessage id={errorMessageId} errorMsg={errorMsg} hidden={hideErrorMessage} {...tid.errorMsg} />
             )}
-            {errorInTooltip && errorMsg && !hideErrorMessage && (
-              <span css={Css.df.aic.pl1.fs0.$}>
-                <Icon icon="error" color={Palette.Red600} tooltip={errorMsg} />
-              </span>
+            {helperText && <HelperText helperText={helperText} {...tid.helperText} />}
+          </>
+        )}
+      </div>
+      {/* alert and helper text for "left" labelStyle */}
```

Comment:
There's probably a more elegant way of doing this, but even small changes to the current approach would result in several dom diffs in different places, so this is the _make sure we're don't break anything_ approach